src/unitxt/metrics.py

```diff
@@ -224,134 +253,207 @@ def disable_confidence_interval_calculation(self):
         pass
 
 
+class Aggregator(Artifact):
+    """Aggregates over a list of instances, updating the score/global field of each of them by a dictionary of scores."""
+
+    score_names: List[str] = None
+
+    # non trivial only for grouper_aggregator
+    aggregator_based_score_prefix = ""
+
+    # computes global score, and accordingly updates instance["score"]["global"] of each instance
+    @abstractmethod
+    def aggregate(self, instances: List[Dict[str, Any]]) -> Dict[str, Any]:
+        pass
+
+    def update_global_score(self, instances: List[Dict[str, Any]], gs: Dict[str, Any]):
+        for instance in instances:
+            if "global" not in instance["score"]:
+                instance["score"]["global"] = {}
+            instance["score"]["global"].update(gs)
+
+    # prefix and add "score" and "score_name"
+    def prefix_global_scores(
+        self, gs: Dict[str, Any], metric: Metric
+    ) -> Dict[str, Any]:
+        full_prefix = self.aggregator_based_score_prefix + metric.score_prefix
+        new_gs = {}
+        new_gs["score_name"] = full_prefix + metric.main_score
+        new_gs["score"] = gs[metric.main_score]
+        if metric.main_score + "_ci_low" in gs:
+            new_gs["score_ci_low"] = gs[metric.main_score + "_ci_low"]
+        if metric.main_score + "_ci_high" in gs:
+            new_gs["score_ci_high"] = gs[metric.main_score + "_ci_high"]
+
+        for score_name, score in gs.items():
+            new_gs[full_prefix + score_name] = score
+
+        return new_gs
+
+    # trivial for all but grouperaggregator
+    def instances_to_sample_from_and_sample_aggregator(
+        self, instances: List[Dict[str, Any]]
+    ) -> Tuple[List[Dict[str, Any]], Any]:
+        return (instances, self)
+
```

Comment:
```suggestion
     def __call__(self, instances: List[Dict[str, Any]]):
          global_score = self.aggregate(instances)
          global_score = self.global_score(global_score)
          self.update_global_score(instacnces)
```
The aggregator should be in charge of all the affected raised from the aggregation

Reply:
It should also supply hints for the CI computer.  So it can not be simply encapsulated.